src/apps/answers/service.py

```diff
@@ -1922,9 +1923,38 @@ async def is_flow_finished(self, submit_id: uuid.UUID, answer_id: uuid.UUID) ->
         return self._is_activity_last_in_flow(applet_full, activity_id, flow_id)
 
     async def create_report(
-        self, submit_id: uuid.UUID, answer_id: uuid.UUID | None = None
+        self,
+        submit_id: uuid.UUID,
+        answer_id: uuid.UUID | None = None,
+        activity_id: uuid.UUID | None = None,
+        report_name: str | None = None,
     ) -> ReportServerResponse | None:
         answers = await AnswersCRUD(self.answers_session).get_by_submit_id(submit_id, answer_id)
+        subscale_table_data: list[SubScaleLookupTable] | None = None
+        subscale_name = ""
+        scoring_type = ""
```

Comment:
Don't need to declare subscale_name here, it is not used other place than line 1953 and initializing variable with None is batter than empty string

Reply:
I usually define variables in the area of the code where they are required. Added subscale_name, scoring_type at the begining based on a previous comment but for me it makes more sense to call the directly like: 
`        answers = await AnswersCRUD(self.answers_session).get_by_submit_id(submit_id, answer_id)
        subscale_table_data: list[SubScaleLookupTable] | None = None

        if activity_id:
            activity_data = await ActivitiesCRUD(self.session).get_by_id(activity_id)
            reports = activity_data.scores_and_reports.get("reports", [])
            if not reports:
                return None  # Early return if no reports are found

            selected_report = next((report for report in reports if report["name"] == report_name), reports[0])

            scoring_type = selected_report.get("scoring_type") <- HERE **********************
            if scoring_type == "score":
                subscale_name = selected_report.get("subscale_name") <- HERE **********************
                subscales = activity_data.subscale_setting.get("subscales", [])
                subscale_table_data = next(
                    (
                        subscale.get("subscale_table_data", [])
                        for subscale in subscales
                        if subscale.get("name") == subscale_name
                    ),
                    [],
                )`